sdk_extensions/logging/support/src/main/java/io/opentelemetry/sdk/logging/export/BatchLogProcessor.java

```diff
@@ -0,0 +1,401 @@
+import java.util.Map;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.RejectedExecutionHandler;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.logging.Logger;
+
+public class BatchLogProcessor implements LogProcessor {
+  private static final String WORKER_THREAD_NAME =
+      BatchLogProcessor.class.getSimpleName() + "_WorkerThread";
+  private static final String TIMER_THREAD_NAME =
+      BatchLogProcessor.class.getSimpleName() + "_TimerThread";
+
+  private final Worker worker;
+  private final Thread workerThread;
+
+  private BatchLogProcessor(
+      int maxQueueSize,
+      long scheduleDelayMillis,
+      int maxExportBatchSize,
+      long exporterTimeout,
+      LogExporter logExporter) {
+    this.worker =
+        new Worker(
+            maxQueueSize, scheduleDelayMillis, maxExportBatchSize, exporterTimeout, logExporter);
+    this.workerThread = new DaemonThreadFactory(WORKER_THREAD_NAME).newThread(worker);
+    this.workerThread.start();
+  }
+
+  public static Builder builder(LogExporter logExporter) {
+    return new Builder(logExporter);
+  }
+
+  @Override
+  public void addLogRecord(LogRecord record) {
+    worker.addLogRecord(record);
+  }
+
+  @Override
+  public void shutdown() {
+    workerThread.interrupt();
+    worker.shutdown();
+  }
+
+  @Override
+  public void forceFlush() {
+    worker.forceFlush();
+  }
+
+  private static class Worker implements Runnable {
+    static {
+      Meter meter = OpenTelemetry.getMeter("io.opentelemetry.sdk.logging");
+      LongCounter logProcessorErrors =
+          meter
+              .longCounterBuilder("logProcessorErrors")
+              .setUnit("1")
+              .setDescription("Number of errors encountered while processing logs")
+              .build();
+      Labels.Builder builder =
+          Labels.of("logProcessorType", BatchLogProcessor.class.getName()).toBuilder();
+      exporterFailureCounter =
+          logProcessorErrors.bind(builder.setLabel("errorType", "exporter failure").build());
+      exporterBusyCounter =
+          logProcessorErrors.bind(builder.setLabel("errorType", "exporter busy").build());
+      droppedRecordCounter =
+          logProcessorErrors.bind(
+              builder.setLabel("errorType", "dropped record - queue full").build());
+    }
+
+    private static final BoundLongCounter exporterFailureCounter;
+    private static final BoundLongCounter exporterBusyCounter;
+    private static final BoundLongCounter droppedRecordCounter;
+    private static final Logger logger = Logger.getLogger(Worker.class.getName());
+
+    private final Object monitor = new Object();
```

Comment:
We recently refactored BatchSpanProcessor. It'd be nice if we could update to follow the same pattern.